ragatouille/models/index.py

```diff
@@ -201,13 +180,69 @@ def build(
 
         # Instruct colbert-ai to disable forking if nranks == 1
         self.config.avoid_fork_if_possible = True
-        indexer = Indexer(
-            checkpoint=checkpoint,
-            config=self.config,
-            verbose=verbose,
-        )
-        indexer.configure(avoid_fork_if_possible=True)
-        indexer.index(name=index_name, collection=collection, overwrite=overwrite)
+
+        # Monkey-patch colbert-ai to avoid using FAISS
+        monkey_patching = False
+        if len(collection) < 500000 and kwargs.get("use_faiss", False) is False:
+            print(
+                "---- WARNING! You are using PLAID with an experimental replacement for FAISS for greater compatibility ----"
+            )
+            print("This is a behaviour change from RAGatouille 0.8.0 onwards.")
+            print(
+                "This works fine for most users, but is slower than FAISS and slightly more approximate."
+            )
+            print(
+                "If you're confident with FAISS working issue-free on your machine, pass use_faiss=True to revert to the FAISS-using behaviour."
+            )
+            print("--------------------")
+            CollectionIndexer._original_train_kmeans = CollectionIndexer._train_kmeans
```

Comment:
We could yeah! This was mostly for the convenience of checking on `hasattr` later on, but it might be better practice to set it to a new object instead. I'll change it

Reply:
Mentioned on Discord but I'm actually thinking this is a relatively sane way of doing it, because we need to keep it alive for the entirety of the session -- we're monkey-patching the `colbert-ai` indexer itself and we want to be able to revert anytime someone needs to use `faiss`, so local variables wouldn't cut it.

Reply:
right yea, makes sense!

In this case, can we assign the faiss and non-faiss k-means functions as class attributes of `PLAIDModelIndex`. Then, at build time, we can just toggle between them (to set CollectionIndexer._train_k_means) based on the monkey_patch flag. Wdyt?

Reply:
Curious to get your thoughts here too @jlscheerer !

Reply:
I like the idea of having it be a class attribute on `PLAIDModelIndex` and toggling it on use (+ persisting the flag). This would perhaps provide more consistent behaviour when rebuilding/adding to an already persisted index (e.g., if we decide to rebuild as part of `add_to_index`).

Reply:
Great suggestion! Implemented now,